### ISSUE TITLE
MySQL Changelog Timestamps are Incorrect

### ISSUE THREAD

**Issue by jon-whit** [openfga/openfga#1177]
**Summary**

Describe the bug here.

- Steps to reproduce

Write a tuple, sleep, and write another tuple.

```
Write(document:1#viewer@user:jon)
time.Sleep(10 * time.Millisecond)
Write(document:2#viewer@user:bob)
```

- What did you expect to happen?

ReadChanges should return a changelog with two items, and the second item should have a timestamp that is about 10ms later than the first.

- What actually happened?
The second changelog item had a timestamp with the same second resolution/precision as the first.

**OpenFGA version or commit**
<= v1.3.7

**Store data**

If applicable, provide information about your authorization model, tuples, and the Check or ListObjects calls you're making. For example:

```yaml
model_file: |
  model
    schema 1.1
  type user
  type document
    relations
      define viewer: [user]
```

**Other data**

- How are you running OpenFGA? (As a binary, in Docker, in Kubernetes)

Binary, but the others are impacted as well.

- What datastore are you using? (In memory, MySQL, Postgres)

MySQL

- Are you running OpenFGA with any configuration overrides or with any of the flags mentioned in `./openfga run --help`?

Using the defaults.

- Do you have any logs or traces that could help us debug the problem?

N/A

Relevant links:
* https://github.com/openfga/openfga/blob/main/pkg/storage/mysql/mysql.go#L195
* https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_now (`NOW()` is second precision)

**Comment by jon-whit** [comment]
Using `NOW()` in both the Postgres and MySQL implementations is naive. It returns the timestamp at which the statement involving the `NOW()` is started, but not the commit timestamp of the transaction. For example:

**txn1** starts first
```
BEGIN;
INSERT INTO table (some_timestamp) VALUES ("NOW()");
pg_sleep(10);
COMMIT;
```

**txn2** starts milliseconds afterwards,
```
BEGIN;
INSERT INTO table (some_timestamp) VALUES ("NOW()");
COMMIT;
```

txn2 commits and therefore is written before txn1, but from the observer that queries by timestamp it will appear that txn1 committed before txn2. 

**tl;dr** - timestamps should be avoided to establish commit order. Atomically incrementing numeric values should be preferrable if order is important but temporality not so much..

The problem is that the `NOW()` function returns the statement timestamp, not the commit timestamp. Depending on timestamp ordering is not safe if you're going to use the `NOW()` function. It's more preferrable to use a autoincrementing revision or some numerical column that is strictly increasing. Such values can be more easily obtained from a `BIGSERIAL` value. However, if a timestamp is important there are some options to get a transaction commit timestamp, but support for it is spotty across different databases, for exapmle I'm not sure MySQL has a commit timestamp precisely, but Postgres has https://www.postgresql.org/docs/current/runtime-config-replication.html#GUC-TRACK-COMMIT-TIMESTAMP and https://www.postgresql.org/docs/current/functions-info.html#FUNCTIONS-COMMIT-TIMESTAMP

**Comment by miparnisari** [comment]
I don't see how your second comment applies to this issue.

We need to use change `inserted_at=NOW()` to `inserted_at=convert_to_ts(ULID)`
